chichiwang/chichiwang.dev
# Add site color palette: Panda

## Description
Linked to Issue: #4
Add a color palette (called Panda) which consists of an off-white, off-black, and two shades of grey that will comply with [WCAG 2.0 color contrast AA requirements](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-contrast).

Globally, background color is set to white, text is set to grey. Links are set to black. The contrast between links and text is 1.88, which does not meet the [WCAG AA compliance guideline for links](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-without-color) - links will need to be styled to differentiate them further from body text (underline, styles, etc).

A new component `LogoBlock` has been created to house the site logo, centered in a block with appropriate margins. This is to avoid requiring each page from having to individually style the logo using the colors. Every page can drop in the LogoBlock component if desired for use on the page header.

An overflow rule has been added to the page element in the Panda layout to ensure that children with margin rules do not push the page element further down the page.

## Changes
* Component `app/components/Logo/LogoBlock.tsx` added
  * Styles added in `app/components/Logo/styles.module.scss`
* Update `app/global.module.scss` to set body and link colors using the Panda palette
* Set page element overflow to `hidden` in `app/layouts/Panda/styles.module.scss`
* Remove previous `$colors` variables and replace with a map of `panda` colors in `app/var.module.scss`
* Replace specific implementation of `Logo` with new `LogoBlock` component in `pages/ErrorPage/index.tsx`
* Replace specific implementation of `Logo` with new `LogoBlock` component in `pages/NotFound/index.tsx`
* Replace specific implementation of `Logo` with new `LogoBlock` component in `pages/Resume/index.tsx`

## Steps to QA
* Pull down and switch to this branch
* Run the app and verify in browser
  * The new colors are applied correctly
  * The logo appears correct on every page (`/`, `/error`, `/notfound`)
